scipy/stats/tests/test_morestats.py

```diff
@@ -193,8 +193,8 @@ def test_nan_input(self):
         shapiro_test = stats.shapiro(x)
         assert_equal(w, np.nan)
         assert_equal(shapiro_test.statistic, np.nan)
-        assert_almost_equal(pw, 1.0)
-        assert_almost_equal(shapiro_test.pvalue, 1.0)
+        assert_almost_equal(pw, np.nan)
+        assert_almost_equal(shapiro_test.pvalue, np.nan)
```

Comment:
I don't think it ever made sense for the p-value to be 1 when the statistic is `nan`.